WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -433,6 +437,38 @@ private void initializePostObject() {
                     AztecEditorFragment.getMediaMarkedUploadingInPostContent(this, mPost.getContent());
             Collections.sort(mMediaMarkedUploadingOnStartIds);
             mIsPage = mPost.isPage();
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    // run this purge in the background to not delay Editor initialization
+                    purgeMediaToPostAssociationsIfNotInPostAnymore();
+                }
+            }).start();
+        }
+    }
+
+    private void purgeMediaToPostAssociationsIfNotInPostAnymore() {
+        ArrayList<MediaModel> allMedia = new ArrayList<>();
+        Set<MediaModel> failedMedia = mUploadStore.getFailedMediaForPost(mPost);
+        Set<MediaModel> completedMedia = mUploadStore.getCompletedMediaForPost(mPost);
+        Set<MediaModel> uploadingMedia = mUploadStore.getUploadingMediaForPost(mPost);
+        allMedia.addAll(failedMedia);
+        allMedia.addAll(completedMedia);
+        allMedia.addAll(uploadingMedia);
+
+        if (allMedia != null && !allMedia.isEmpty()) {
```

Comment:
Niptick - `allMedia` cannot be `null` here.

Reply:
👍 removed unnecessary check in cb62466